### PR TITLE
[https://nvbugspro.nvidia.com/bug/6778095] Fix fused P-QDQ to respect disabled quantization during calibration

### PR DESCRIPTION
**Pull request by yingguo-trt**
### What does this PR do?

Type of change: Bug fix

During max calibration, `enable_stats_collection()` calls `disable_quant()`, which sets `_if_quant=False`. The fused causal P-QDQ attention paths bypass `TensorQuantizer.forward()` and previously selected the Triton/Kitchen path from the configured enabled state alone, so P quant-dequant could still execute while quantization was inactive.

This change:

- enters the Triton P-QDQ path only when `p_bmm_quantizer._if_quant` is true;
- bypasses all fused P-QDQ paths when the P quantizer is disabled or quantization is inactive;
- adds a focused regression test covering causal attention with `disable_quant()`.

### Usage

N/A. This restores the existing `disable_quant()` contract and does not introduce a new API.

### Testing

- `pytest tests/unit/torch/quantization/plugins/test_attention_quant.py`: 9 passed
- Targeted pre-commit checks passed
- Reproduced with the exact same ModelOpt 0.47.0rc1 wheel on both sides on B300: [module regression build #121](http://dlswqa-nas.nvidia.com:18880/view/yiguo/job/modelopt-quant-module/121/)
- Controlled B300 isolation passed when only the P-BMM quantizers were hard-disabled during calibration, and also passed when the existing single Triton attention configuration was forced

The fixed-commit GPU end-to-end validation is still pending. The confirmed impact is incorrect calibration behavior plus unstable quantizer state and module outputs; downstream benchmark accuracy impact has not been established.

### Before your PR is "*Ready for review*"

- Is this change backward compatible?: ✅
- If you copied code from any other sources or added a new PIP dependency, did you follow guidance in `CONTRIBUTING.md`: N/A
- Did you write any new necessary tests?: ✅
- Did you update [Changelog](https://github.com/NVIDIA/Model-Optimizer/blob/main/CHANGELOG.rst)?: N/A
- Did you get Claude approval on this PR?: N/A

### Additional Information

The failure was isolated to fused P-QDQ runtime-state dispatch. Quantizer topology and configuration were identical in the failing same-wheel comparison.

## Summary by CodeRabbit

- **Bug Fixes**
  - Corrected attention behavior so disabled or non-quantizing configurations use the original attention implementation instead of the quantized path.
  - Preserved optimized quantized attention when quantization is enabled.
  - Preserved fallback support for fused attention where applicable.
  - Ensured attention masks remain unchanged when using the original attention implementation.

- **Tests**
  - Added coverage for enabled and disabled quantization states, attention-mask handling, fallback selection, and result consistency.

### CODE REVIEW

File: modelopt/torch/quantization/plugins/huggingface.py [hunk 1]
```diff
@@ -247,8 +247,15 @@ def _quantized_attention(
...
-                p_qdq, query_states, key_states, value_states, **kwargs
+            if self.p_bmm_quantizer._if_quant:
+                return self._triton_qdq_attention(
+                    p_qdq, query_states, key_states, value_states, **kwargs
+                )
+
+        # Fused P-QDQ paths bypass TensorQuantizer.forward().
+        if not self.p_bmm_quantizer.is_enabled or not self.p_bmm_quantizer._if_quant:
+            return original_attention_interface(
+                self, query_states, key_states, value_states, *args, **kwargs
```
[the remark this answers is not in the document]
Comment (yingguo-trt) [reply modelopt/torch/quantization/plugins/huggingface.py:260]: Fixed in 77c4b94ad. The fallback now removes the normalized keyword mask before forwarding the original positional arguments. The existing regression test now passes a sentinel mask positionally to a callback with an explicit attention_mask parameter, so duplicate binding is exercised directly. The focused test file passes 9/9 and all targeted pre-commit hooks pass.

**Review by cjluo-nv — COMMENTED** [review]
> _Bot review (gpt-6-astra) — DM the bot to share feedback._

Nudge: the runtime-state guard looks correct, but Kitchen bypass coverage and fixed-commit GPU validation remain incomplete.

Needs action:
- [ ] Extend `test_attention_quant.py` to cover `disable()` and `disable_quant()` with Kitchen available, including an already-initialized Kitchen callable; assert neither initialization nor execution occurs while disabled and dispatch resumes after re-enabling.
- [ ] Run the pending fixed-commit GPU calibration regression and record the result in the PR body.

No action needed:
- The new FP8 dispatch test covers active/inactive quantization and preserves the positional attention mask. Existing tests are unchanged.

**Review by sychen52 — APPROVED** [review]

**Comment by sychen52** [comment]
Good cache. Thanks!

**Review by Edwardf0t1 — APPROVED** [review]
LGTM

**Comment by yingguo-trt** [comment]
@cjluo-nv Addressed both follow-ups in `693f7fba8` after rebasing onto the current `upstream/main`:

- Added one parameterized Kitchen dispatch test covering `disable()` before initialization and `disable_quant()` with an already-initialized callable. Both cases assert that Kitchen is neither initialized nor executed while inactive and that dispatch resumes after the matching re-enable operation.
- Updated the PR body with three independent fixed-code B300 passes: [Jenkins #123](http://dlswqa-nas.nvidia.com:18880/job/modelopt-quant-module/123/), [Jenkins #124](http://dlswqa-nas.nvidia.com:18880/job/modelopt-quant-module/124/), and [Jenkins #125](http://dlswqa-nas.nvidia.com:18880/job/modelopt-quant-module/125/).
- The focused attention test file passes 12/12, and the targeted pre-commit checks pass.

Could an authorized vetter please run `/ok to test 693f7fba8a5e038dbc520e793b669b6be15eca9e` for the NVIDIA-hosted required checks?

**Review by cjluo-nv — APPROVED** [review]
> _Bot review (gpt-6-astra) — DM the bot to share feedback._

Approve: the prior correctness and validation concerns are resolved, with focused regression coverage and three reported fixed-code B300 passes.

No action needed:
- :heavy_check_mark: Resolved since the last review: positional-mask duplicate binding, Kitchen bypass/re-enable coverage across all four runtime-state cases, and fixed-code GPU validation.
- Existing tests are unchanged; the added tests reuse the shared attention fixture.
